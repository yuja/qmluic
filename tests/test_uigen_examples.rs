pub mod common;

#[test]
fn test_gadget_properties() {
    insta::assert_snapshot!(common::translate_file("examples/GadgetProperties.qml").unwrap());
}

#[test]
fn test_item_views() {
    insta::assert_snapshot!(common::translate_file("examples/ItemViews.qml").unwrap());
}

#[test]
fn test_layout_flow() {
    insta::assert_snapshot!(common::translate_file("examples/LayoutFlow.qml").unwrap());
}

#[test]
fn test_main_window() {
    insta::assert_snapshot!(common::translate_file("examples/MainWindow.qml").unwrap());
}

#[test]
fn test_settings_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/SettingsDialog.qml").unwrap());
}

#[test]
fn test_static_item_model() {
    insta::assert_snapshot!(common::translate_file("examples/StaticItemModel.qml").unwrap());
}

#[test]
fn test_various_layouts() {
    insta::assert_snapshot!(common::translate_file("examples/VariousLayouts.qml").unwrap());
}

#[test]
fn test_customwidget_common_my_dialog_button_box() {
    insta::assert_snapshot!(common::translate_file(
        "examples/customwidget/common/MyDialogButtonBox.qml"
    )
    .unwrap());
}

#[test]
fn test_customwidget_main_dialog() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/MainDialog.qml").unwrap()
    );
}

#[test]
fn test_customwidget_settings_form() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/SettingsForm.qml").unwrap()
    );
}

#[test]
fn test_thg_hg_email_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/thg/HgEmailDialog.qml").unwrap());
}
