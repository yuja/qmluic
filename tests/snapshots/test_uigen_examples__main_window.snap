---
source: tests/test_uigen_examples.rs
expression: ui_xml
---
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="root">
  <property name="geometry">
   <rect>
    <height>600</height>
    <width>800</width>
    <x>0</x>
    <y>0</y>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="vboxLayout">
    <item>
     <layout class="QFormLayout" name="formLayout">
      <item column="0" row="0">
       <widget class="QLabel" name="label">
        <property name="buddy">
         <cstring>fileNameEdit</cstring>
        </property>
        <property name="text">
         <string>&amp;Source</string>
        </property>
       </widget>
      </item>
      <item column="1" row="0">
       <widget class="QComboBox" name="fileNameEdit">
        <item>
         <property name="text">
          <string notr="true">MainWindow.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">BindingLoop.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">HgEmailDialog.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">ItemViews.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">LayoutFlow.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">SettingsDialog.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">StaticItemModel.qml</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">VariousLayouts.qml</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       </sizepolicy>
      </property>
      <widget class="QPlainTextEdit" name="sourceEdit">
       <property name="font">
        <font>
         <family notr="true">Monospace</family>
        </font>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="textInteractionFlags">
        <set>Qt::TextSelectableByMouse|Qt::TextSelectableByKeyboard</set>
       </property>
      </widget>
      <widget class="QStackedWidget" name="formStack">
       <widget class="QWidget" name="widget">
       </widget>
       <widget class="BindingLoop" name="bindingLoop">
       </widget>
       <widget class="HgEmailDialog" name="hgEmailDialog">
       </widget>
       <widget class="ItemViews" name="itemViews">
       </widget>
       <widget class="LayoutFlow" name="layoutFlow">
       </widget>
       <widget class="SettingsDialog" name="settingsDialog">
       </widget>
       <widget class="StaticItemModel" name="staticItemModel">
       </widget>
       <widget class="VariousLayouts" name="variousLayouts">
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <addaction name="menu"/>
   <addaction name="menu1"/>
   <addaction name="menu3"/>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Open"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu1">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="action_Undo"/>
    <addaction name="action_Redo"/>
   </widget>
   <widget class="QMenu" name="menu3">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
    <addaction name="separator"/>
    <addaction name="menu2"/>
    <action name="action_About">
     <property name="icon">
      <iconset>
       <normaloff>app.png</normaloff>
      </iconset>
     </property>
     <property name="text">
      <string>&amp;About</string>
     </property>
    </action>
    <widget class="QMenu" name="menu2">
     <property name="title">
      <string>&amp;Developer tools</string>
     </property>
     <addaction name="action1"/>
     <action name="action1">
      <property name="text">
       <string>&amp;Console</string>
      </property>
     </action>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <addaction name="action_Open"/>
   <addaction name="separator"/>
   <addaction name="action_Undo"/>
   <addaction name="action_Redo"/>
   <addaction name="separator"/>
   <addaction name="action_About"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset theme="document-open">
    </iconset>
   </property>
   <property name="shortcut">
    <enum>QKeySequence::Open</enum>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="shortcut">
    <string notr="true">Ctrl+X, Ctrl+C</string>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="icon">
    <iconset theme="edit-undo">
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-redo">
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="actionFoo">
   <property name="text">
    <string>Foo</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BindingLoop</class>
   <extends>QWidget</extends>
   <header>bindingloop.h</header>
  </customwidget>
  <customwidget>
   <class>HgEmailDialog</class>
   <extends>QDialog</extends>
   <header>hgemaildialog.h</header>
  </customwidget>
  <customwidget>
   <class>ItemViews</class>
   <extends>QWidget</extends>
   <header>itemviews.h</header>
  </customwidget>
  <customwidget>
   <class>LayoutFlow</class>
   <extends>QWidget</extends>
   <header>layoutflow.h</header>
  </customwidget>
  <customwidget>
   <class>SettingsDialog</class>
   <extends>QDialog</extends>
   <header>settingsdialog.h</header>
  </customwidget>
  <customwidget>
   <class>StaticItemModel</class>
   <extends>QWidget</extends>
   <header>staticitemmodel.h</header>
  </customwidget>
  <customwidget>
   <class>VariousLayouts</class>
   <extends>QWidget</extends>
   <header>variouslayouts.h</header>
  </customwidget>
 </customwidgets>
</ui>

